[API design] Add Fleet-maintained apps via YAML (GitOps) (#27204)
fleetdm/fleet: docs/REST API/rest-api.md

- [List team policies](#list-team-policies)
- [Count policies](#count-policies)
- [Count team policies](#count-team-policies)

> - [Get policy by ID](#get-policy-by-id)
> - [Get team policy by ID](#get-team-policy-by-id)

Reviewer: I think these naming changes are also an unintentional result of switching the branch (we recently removed "by ID" on `main`).

But hey, it's not 200+ files worth of changes so I'd call it a win! 😅 
```suggestion
- [Get policy](#get-policy)
- [Get team policy](#get-team-policy)
```

---

> ### Get team policy by ID

Reviewer: ```suggestion
### Get team policy
```

| Name | Type | In | Description |
| ---- | ---- | -- | ----------- |
| id   | integer | path | **Required.** The Fleet-maintained app's ID. |

> | fleet_maintained_app_id   | integer | body | **Required.** The ID of Fleet-maintained app. |

Reviewer 2: ```suggestion
| slug   | integer | body | **Required.** The slug of Fleet-maintained app. You can find all the slugs [here in GitHub](https://github.com/fleetdm/fleet/blob/main/ee/maintained-apps/outputs/apps.json).|
```

> | slug   | integer | body | **Required.** The slug of Fleet-maintained app. You can find all the slugs [here in GitHub](https://github.com/fleetdm/fleet/blob/main/ee/maintained-apps/outputs/apps.json).|

Reviewer 2: ```suggestion
```

@eugkuo I don't think we need to add `slug` as a request parameter. The API user will request a Fleet-maintained app using `id`. `slug` will just show up in the response.

Author: @noahtalerman OKee. I'll remove this.

{
  "fleet_maintained_app": {
    "id": 1,

>     "fleet_maintained_app_id": 3,

Reviewer 2: ```suggestion
    "slug": "adobe-acrobat-reader/darwin",
```
